internal/state/graph/httproute.go

```diff
@@ -405,20 +406,31 @@ func findAcceptedHostnames(listenerHostname *v1beta1.Hostname, routeHostnames []
 		if hostname == "" {
 			return true
 		}
-		return string(h) == hostname
+
+		routeHost := string(h)
+		return routeHost == hostname || wildcardMatch(hostname, routeHost) || wildcardMatch(routeHost, hostname)
 	}
 
 	var result []string
 
 	for _, h := range routeHostnames {
 		if match(h) {
-			result = append(result, string(h))
+			if len(hostname) > len(h) {
```

Comment:
what if h is `*.example.com` and hostname is `f.example.com`. In this case, `f.example.com` should be in the config.